source-code/core/src/lint/linter.ts

```diff
@@ -13,46 +13,39 @@ import type {
 const getResourceForLanguage = (resources: Resource[], language: string) =>
 	resources.find(({ languageTag }) => languageTag.name === language)
 
-export const getLintRulesFromConfig = (config: Config) => (config.lint?.rules || []).flat()
-
-export const lint = async (config: Config, env: EnvironmentFunctions) => {
-	const { referenceLanguage, languages, readResources } = config
-
-	const lintRules = getLintRulesFromConfig(config)
-	if (!lintRules.length) {
+export const lint = async (args: {
+	config: Pick<Config, "lint" | "languages" | "referenceLanguage">
+	resources: Resource[]
+}) => {
+	const { referenceLanguage, languages, lint } = args.config
+	if (lint === undefined || lint.rules.length === 0) {
 		console.warn("No lint rules specified. Aborting ...")
 		return
 	}
-
-	const resources = await readResources({ config })
-
-	const reference = getResourceForLanguage(resources, referenceLanguage)
+	const reference = getResourceForLanguage(args.resources, referenceLanguage)
 
 	await Promise.all(
-		lintRules.map((lintRule) =>
+		lint?.rules.flat().map((lintRule) =>
```

Comment:
At this point we are sure that `lint` is defined
```suggestion
		lint.rules.flat().map((lintRule) =>
```